### PR TITLE
Fix test issues

### PR DESCRIPTION
**Pull request by auroraslb**
Fix issues discovered in the EDTT tests.

### CODE REVIEW

**Comment by joerchan** [comment]
@auroraslb Could you add some more information in the commit changes about why the change and why it is needed?

File: src/components/basic_commands.py [hunk 1]
```diff
@@ -1879,13 +1879,17 @@ def le_read_peer_resolvable_address(transport, idx, PeerIdAddrType, AVal, to):
...
-    
-    if ( 11 != len(packet) ):
-        raise Exception("LE Read Peer Resolvable Address command failed: Response too short (Expected %i bytes got %i bytes)" % (11, len(packet)));
+    packet = transport.recv(idx, 5, to);
+    RespCmd, RespLen, status = struct.unpack('<HHB', packet);
+
+    if ( status == 0 ):
+        packet = transport.recv(idx, 6, to);
+        if ( 6 != len(packet) ):
+            raise Exception("LE Read Peer Resolvable Address command failed: Response too short (Expected %i bytes got %i bytes)" % (11, len(packet)));
```
Comment (joerchan) [inline src/components/basic_commands.py:1888]:
```suggestion
            raise Exception("LE Read Peer Resolvable Address command failed: Response too short (Expected %i bytes got %i bytes)" % (6, len(packet)));
```

File: src/components/basic_commands.py [hunk 2]
```diff
@@ -1904,13 +1908,17 @@ def le_read_local_resolvable_address(transport, idx, PeerIdAddrType, AVal, to):
...
-    
-    if ( 11 != len(packet) ):
-        raise Exception("LE Read Local Resolvable Address command failed: Response too short (Expected %i bytes got %i bytes)" % (11, len(packet)));
+    packet = transport.recv(idx, 5, to);
+    RespCmd, RespLen, status = struct.unpack('<HHB', packet);
+
+    if ( status == 0 ):
+        packet = transport.recv(idx, 6, to);
+        if ( 6 != len(packet) ):
+            raise Exception("LE Read Local Resolvable Address command failed: Response too short (Expected %i bytes got %i bytes)" % (11, len(packet)));
```
Comment (joerchan) [inline src/components/basic_commands.py:1917]:
```suggestion
            raise Exception("LE Read Local Resolvable Address command failed: Response too short (Expected %i bytes got %i bytes)" % (6, len(packet)));
```

**Review by joerchan — CHANGES_REQUESTED** [review]
The test changes look correct to me.

File: src/components/basic_commands.py [hunk 3]
```diff
@@ -1904,13 +1908,17 @@ def le_read_local_resolvable_address(transport, idx, PeerIdAddrType, AVal, to):
...
     transport.send(idx, cmd);
     
-    packet = transport.recv(idx, 11, to);
-    
-    if ( 11 != len(packet) ):
-        raise Exception("LE Read Local Resolvable Address command failed: Response too short (Expected %i bytes got %i bytes)" % (11, len(packet)));
+    packet = transport.recv(idx, 5, to);
+    RespCmd, RespLen, status = struct.unpack('<HHB', packet);
+
+    if ( status == 0 ):
```
Comment (kruithofa) [inline src/components/basic_commands.py:1914]: This causes amongst others BV-14-C to fail when running against the Zephyr LL with status equals 2 (unknown connection identifier IIRC).
Further investigation is required on why.

**Review by kruithofa — CHANGES_REQUESTED** [review]

**Review by joerchan — APPROVED** [review]

File: src/components/basic_commands.py (hunk 3 above)
Comment (joerchan) [reply src/components/basic_commands.py:1914]: @kruithofa Are you investigating this?

Comment (kruithofa) [reply src/components/basic_commands.py:1914]: My bad, the full name is LL/DDI/SCN/BV-14-C.

Yes, I am investigating this. I had a look at it, and need to discuss further with Aurora

Comment (kruithofa) [reply src/components/basic_commands.py]: The offending commit has been taken out from this PR, and issue #21 was created

**Review by heri-developer — APPROVED** [review]

**Review by kruithofa — APPROVED** [review]

**Pull request merged**